### DISCUSSION TITLE
condition if not working

### DISCUSSION THREAD

**Discussion by andressalhenriques** [maizzle/framework#923] [General]
I'm receiving my variables from another file that uses this template.

so if I do that, I can get the value title
```
<h3 align="center" class="text-parfin-dark font-semibold mt-8">
  @{{ $title }}
</h3>

```
However, when I put a condition it did not work.
```
<if condition="$title">
  <p align="center" class="mt-8 text-parfin-darker leading-[25px]">
     @{{ $title}}
 </p>
</if>
```

**Comment by cossssmin** [comment]
How is `$title` defined?

**Reply by andressalhenriques** [reply]
like this:

```
` $title: 'Is Almost Ready To Use',
  $description: 'You account was created. To finish the setup please follow the steps bellow.',
 $description1: 'Texto 2',`
```
I call my lambda that gets the template and changes the variables, so my template gets these 3 variables.like this:

Do you guys have some documentation with examples that get value from variables different from page.something?

**Reply by cossssmin** [reply]
Front Matter variables are accessible through the `page` object:

https://maizzle.com/docs/templates#front-matter

So your condition needs to be `page.$title`:

```html
<if condition="page.$title">
  <p align="center" class="mt-8 text-parfin-darker leading-[25px]">
     @{{ $title}}
 </p>
</if>
```

Also note that @{{ $title }} will not evaluate the expression and output as `$title`.

If you need to output the value of the `$title` variable that you've set in Front Matter, use `{{ page.$title }}`.

**Reply by andressalhenriques** [reply]
I see, I don't know if I understand well. However I didn't set in at the begging of the file(front matter), I receive it from another file. So I already try to get page.title but I got undefined.

This work and I can see the correct value on my email.
`<p align="center" class="m-0 text-darker leading-[25px] mt-8">
          @{{ $title }}
        </p>`

**Reply by a-smily** [reply]
How to use ```<if condition="$title">``` if the variable is not set with Front Matter ?

 ```
<if condition="$title">
<title>@{{ $title }}</title>
</if>
```

**Reply by cossssmin** [reply]
how is the variable set then?

**Reply by a-smily** [reply]
I work with Laravel notifications and the variable is inserted at runtime.
Something like this:
```
$url = $this->verificationUrl($notifiable);

$mailData = [
    'page_title' => Lang::get('Confirm your email address'),
    'page_preheader' => Lang::get('Please confirm your email address in order to activate your account.'),
    'verificationUrl' => $url,
    'user_name' => $this->user->name,
    'app_url' => config('app.url'),
    'app_name' => app(GeneralSettings::class)->app_name,
];

return (new MailMessage)
    ->from(app(GeneralSettings::class)->default_email_from_adress, app(GeneralSettings::class)->default_email_from_name) 
    ->replyTo(app(GeneralSettings::class)->default_email_from_adress, app(GeneralSettings::class)->default_email_from_name)
    ->subject(Lang::get('Please verify your email'))
    ->view(['emails.auth.user_verify_email', 'emails.auth.user_verify_email_text'], $mailData);
    }
```

I could still work with blade derectives `@if, @else, @elseif`, but that will get pretty confusing afterwards, if it can be done at all?

**Reply by zkrzyzanowski** [reply]
I'm running into a similar issue as other have described here. `imageUrl` isn't defined in our frontMatter, but dynamically passed in at run time. 

This doesn't return anything  in the UI even though `@{{imageUrl}}` should be false when running it on our dev server.

```html
<if condition="@{{imageUrl}}">
    <img class="w-[50px]" src="@{{imageUrl}}" alt="logo" />
</if>
<else>
    <img
        class="w-[50px]"
        src="images/default-image.png"
        alt="logo"
    />
</else>
```

Any ideas on how to work around this? Basically we need to pass in a dynamic value for images, but want a fallback if nothing is passed in.

**Reply by meganspaulding** [reply]
Having the same issue with dynamically passed in values as @zkrzyzanowski . Would love an update here.

**Reply by cossssmin** [reply]
Expressions cannot be evaluated if they are ignored, so no branch in the statement is reached. Also it should be referenced off the [`page` object](https://maizzle.com/docs/expressions), something like this:

```html
<!-- using triple {{{ }}} braces to avoid escaping URLs -->
<if condition="{{{ page.imageUrl }}}">
    <img class="w-[50px]" src="{{{ page.imageUrl }}}" alt="logo" />
</if>
<else>
    <img
        class="w-[50px]"
        src="images/default-image.png"
        alt="logo"
    />
</else>
```

When `page.imageUrl` is a string it will show the img in the `<if>`, otherwise if it's not defined it'll show the one in the `<else>`.

However if you need to evaluate `imageUrl` on the Laravel side, the `<if>` tags from Maizzle can't work there, you'd need to use the `@if` statements from Blade, maybe something like:

```blade
@if(!empty($imageUrl))
    <img class="w-[50px]" src="{{ $imageUrl }}" alt="logo" />
@else
    <img class="w-[50px]" src="images/default-image.png" alt="logo" />
@endif
```

Hope this helps!
